### PR TITLE
Add ruby 2.6 target for vagrant

### PR DESCRIPTION
**Pull request by gekola**
Add ruby 2.6 and remove 2.3 for latest app-emulation/vagrant and dev-ruby/net-sftp versions.

### CODE REVIEW

**Comment by gyakovlev** [comment]
can't speak for ruby, but last time I checked (couple of weeks ago) net-sftp did not pass tests with ruby2.6 and the errors looked kinda severe, doubt @graaff will approve it.

also NACK conditional ruby_targets deps, that's now how it's done. eclass takes care of that, just set  required version constraints, eclass will handle correct dep with appropriate ruby target.

File: app-emulation/vagrant/vagrant-2.2.4.ebuild [hunk 1]
```diff
@@ -29,6 +29,7 @@ RDEPEND="${RDEPEND}
 ruby_add_rdepend "
 	>=dev-ruby/bcrypt_pbkdf-1.0.0
 	>=dev-ruby/childprocess-0.6.0
+	ruby_targets_ruby26? ( >=dev-ruby/childprocess-1.0.1-r1 )
```
Comment (gyakovlev) [inline app-emulation/vagrant/vagrant-2.2.4.ebuild]: don't do that. eclass and portage already take care of that, as it's not pure `RDEPEND`.
real `RDEPEND` is generated and looks like this
```
...
>=dev-ruby/childprocess-0.6.0[ruby_targets_ruby26]
>=dev-ruby/net-sftp-2.1[ruby_targets_ruby26]
...
```

File: app-emulation/vagrant/vagrant-2.2.4.ebuild [hunk 2]
```diff
@@ -38,6 +39,7 @@ ruby_add_rdepend "
 	<dev-ruby/mime-types-3:*
 	>=dev-ruby/net-ssh-5.1.0:*
 	>=dev-ruby/net-sftp-2.1
+	ruby_targets_ruby26? ( >=dev-ruby/net-sftp-2.1.2-r3 )
```
Comment (gyakovlev) [inline app-emulation/vagrant/vagrant-2.2.4.ebuild]: ditto

File: app-emulation/vagrant/vagrant-2.2.4.ebuild [hunk 3]
```diff
@@ -70,7 +72,7 @@ all_ruby_prepare() {
 	sed -e '/rb-kqueue/d' \
 		-i ${PN}.gemspec || die
 
-	sed -e "s/@VAGRANT_VERSION@/${PV}/g" "${FILESDIR}/${PN}.in" > "${PN}" || die
+	sed -e "s/@VAGRANT_VERSION@/${PV}/g" "${FILESDIR}/${P}.in" > "${PN}" || die
```
Comment (gyakovlev) [inline app-emulation/vagrant/vagrant-2.2.4.ebuild]: add a new revision, no need to introduce package version in file name.

File: dev-ruby/net-sftp/net-sftp-2.1.2-r3.ebuild [hunk 4]
```diff
@@ -1,8 +1,8 @@
-# Copyright 1999-2018 Gentoo Foundation
+# Copyright 1999-2019 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=6
-USE_RUBY="ruby23 ruby24 ruby25"
+USE_RUBY="ruby24 ruby25 ruby26"
```
Comment (gyakovlev) [inline dev-ruby/net-sftp/net-sftp-2.1.2-r3.ebuild]: if it does not pass tests it can't be added.

Comment (gekola) [reply dev-ruby/net-sftp/net-sftp-2.1.2-r3.ebuild]: It does not indeed. But neither does it with any other ruby version.
It looks like the last `net-ssh` version that `net-sftp` passes tests with is 3.2.0, which is not present in the tree currently.
A newer version of `net-sftp` that passes tests with modern `net-ssh` has not been released yet. `net-sftp` v2.1.3.rc3 and v3.0.0.beta1 pass the tests fine with `net-ssh` 5.2.0 and all listed ruby versions, including 2.6. I am not sure if it is OK to bring to rc/betas for gems in the gentoo tree though.

Comment (gyakovlev) [reply dev-ruby/net-sftp/net-sftp-2.1.2-r3.ebuild]: Thanks for detailed info, I'll leave decision on this to @graaff 
if ruby2.6 added I'll be more than happy to merge vagrant stuff and comepletely swtich to 2.6 =)

**Comment by gyakovlev** [comment]
vagrant changes  look good now, this part is approved.
let's wait for resolution on net-sftp

File: dev-ruby/net-sftp/net-sftp-2.1.2-r3.ebuild (hunk 4 above)
Comment (graaff) [reply dev-ruby/net-sftp/net-sftp-2.1.2-r3.ebuild]: It turned out that we had patches backported for that already in net-sftp-2.1.2-r3 but they were not properly applied. This is now fixed in 2.1.2-r4 and that revision also has ruby26 support.

Comment (gekola) [reply dev-ruby/net-sftp/net-sftp-2.1.2-r3.ebuild]: Thanks! I rebased the branch ommiting the `dev-ruby/net-sftp` commit, since it is no longer necessary.

**Comment by gyakovlev** [comment]
merged, thanks for  pushing this forward, now we caught up finally.

**Pull request closed without merging**